### DISCUSSION TITLE
Should correlation coefficient be used as an objective function for optimization?

### DISCUSSION THREAD

**Discussion by ntolley** [jonescompneurolab/hnn-core#1197] [General]
Currently we use RMSE as the primary objective function for optimizing the fit between simulated ERPs.

One of the main differences between RMSE and correlation coefficient is that the input data is **mean centered** and **divided by the standard deviation**  (i.e. z scoring)
<img width="381" height="84" alt="image" src="https://github.com/user-attachments/assets/038aced1-dfee-4556-9132-bcd81ffa65e0" />

In some situations I have found it easier to optimize correlation coefficient rather than RMSE. I want to discuss the scientific strengths/weaknesses of this in the context of HNN

1) Mean centering essentially means we are throwing away information about the baseline. In HNN, we already do a "baseline normalization" so that the dipole starts at zero. Would it be valid to include the baseline as a free parameter for model fitting? This is the question I have the most concerns about.

2) Dividing by the standard deviation isn't really a concern. We already have a scaling factor parameter, so optimizing with correlation essentially translates to using the best fitting scaling factor at every epoch. At the end we get the optimal scaling factor for free. The main risk is that the scaling factor could be any value, so in practice we would need to reject optimized results where the predicted scaling is obviously too large or too small

@stephanie-r-jones @katduecker @carolinafernandezp @rythorpe  @jasmainak I'm curious to hear your opinion on this!

**Comment by rythorpe** [comment]
This is an excellent point @ntolley and a feature we should probably improve. As you point out, we care a lot about the baseline and scaling factor of the final simulated output, so it might make sense to think of the loss function as its own linear regression problem with mean-squared (residual) error as the final output rather than merely a correlation coeff. The important thing from the user's perspective will be to set reasonable default ranges for the offset term and scaling factor and to carefully control the rate at which these parameters changed between optimization trials. I feel like there is some linear control theory behind this problem that could help us here.

**Comment by jasmainak** [comment]
@ntolley I think it's totally consistent with the original vision to offer various options for optimizing the dipoles -- I was imagining a module similar `sklearn.metrics` in `hnn-core` ... including rmse, relative bandpower etc. (documented in API) with an example demoing when one metric might be more appropriate than the other. Bonus points if there is some cross-compatibility with other toolboxes like MNE decoding module, mne-features, or sklearn.
